File: tests/src/com/hendraanggrian/support/utils/test/MimeTypesTest.java
package com.hendraanggrian.support.utils.test;

import android.annotation.SuppressLint;
import android.support.test.runner.AndroidJUnit4;

import com.hendraanggrian.support.utils.content.MimeTypes;

import org.junit.Assert;
import org.junit.Test;
import org.junit.runner.RunWith;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNull;

/**
 * @author Hendra Anggrian (hendraanggrian@gmail.com)
 */
@RunWith(AndroidJUnit4.class)
public class MimeTypesTest {

    @Test
    @SuppressLint("SdCardPath")
    public void guessExtension() throws Exception {
        Assert.assertEquals(MimeTypes.fromUrlOrNull("/sdcard/home/important.txt"), "text/plain");

        assertEquals(MimeTypes.fromExtensionOrNull("txt"), "text/plain");
        assertEquals(MimeTypes.fromExtensionOrNull("jpg"), "image/jpeg");
        assertNull(MimeTypes.fromExtensionOrNull(""));
        assertNull(MimeTypes.fromExtensionOrNull(null));
    }

    @Test
    public void guessUrl() throws Exception {
        assertEquals(MimeTypes.fromUrlOrNull("http://www.somewebsite.com/someimage.jpg"), "image/jpeg");
        assertNull(MimeTypes.fromUrlOrNull(""));
        assertNull(MimeTypes.fromUrlOrNull(null));
    }
}

File: tests/src/com/hendraanggrian/support/utils/test/MimeTypesTest.kt
package com.hendraanggrian.support.utils.test

import android.annotation.SuppressLint
import android.support.test.runner.AndroidJUnit4
import com.hendraanggrian.support.utils.content.extensionToMime
import com.hendraanggrian.support.utils.content.pathToMime
import org.junit.Assert.assertEquals
import org.junit.Assert.assertNull
import org.junit.Test
import org.junit.runner.RunWith

/**
 * @author Hendra Anggrian (hendraanggrian@gmail.com)
 */
@RunWith(AndroidJUnit4::class)
class MimeTypesTest {

    @Test
    @SuppressLint("SdCardPath")
    @Throws(Exception::class)
    fun guessExtension() {
        assertEquals("/sdcard/home/important.txt".pathToMime(), "text/plain")
        assertEquals("txt".extensionToMime(), "text/plain")
        assertEquals("jpg".extensionToMime(), "image/jpeg")
        assertNull("".extensionToMime())
    }

    @Test
    @Throws(Exception::class)
    fun guessUrl() {
        assertEquals("http://www.somewebsite.com/someimage.jpg".pathToMime(), "image/jpeg")
    }
}